nwb_conversion_tools/utils/common_writer_tools.py

```
@@ -0,0 +1,218 @@
+
+
+def reshape_dynamictable(dt, prop_dict, defaults):
+    """
+    Prepares an already existing dynamic table to take custom properties using the add_functions.
+    Parameters
+    ----------
+    dt: DynamicTable
+    prop_dict: dict
+        dict(name=dict(description='',data='', index=bool)
+    defaults: dict
+        default row values for dt columns
+
+    Returns
+    -------
+    default_updated: dict
+    """
+    defaults_updated = defaults
+    if dt is None:
+        return
+    property_default_data = DynamicTableSupportedDtypes
+    for colname in dt.colnames:
+        if colname not in defaults:
+            samp_data = dt[colname].data[0]
+            default_datatype = [proptype for proptype in property_default_data if isinstance(samp_data, proptype)][0]
+            defaults_updated.update({colname: property_default_data[default_datatype]})
+    # for all columns that are new for the given RX, they will
+    for name, des_dict in prop_dict.items():
+        des_args = dict(des_dict)
+        if name not in defaults_updated:
+            # build default junk values for data and add that as column directly later:
+            default_datatype_list = [
+                proptype for proptype in property_default_data if isinstance(des_dict["data"][0], proptype)
+            ][0]
+            des_args["data"] = [property_default_data[default_datatype_list]] * len(dt.id)
+            dt.add_column(name, **des_args)
+
+
+def default_return(value):
+    def return_dec(func):
+        def wrapper(*args, **kwargs):
+            try:
+                out = func(*args, **kwargs)
+                if out is None:
+                    return value
+                else:
+                    return out
+            except Exception as e:
+                if "NoneType" in str(e):
+                    return value
+                else:
+                    return e
+
+        return wrapper
+
+    return return_dec
```

what is the purpose of this? It seems to be silencing all errors. Why would we want that?

---

@Saksham20 please remove

---

> what is the purpose of this? It seems to be silencing all errors. Why would we want that?

I have changed it to raise errors instead of returning them. The only error that we need to deal with is when calling a method on a `NoneType` object (which can happen: https://github.com/catalystneuro/nwb-conversion-tools/pull/230#discussion_r744249753

---

https://github.com/catalystneuro/nwb-conversion-tools/pull/230/commits/d9ea4597f5a70ac143c96c35b922b4c62f8b2a4f